java/client/src/main/java/glide/api/models/commands/StreamOptions.java

```diff
@@ -8,42 +8,77 @@
+/** Optional arguments for {@link StreamBaseCommands} */
+public final class StreamOptions {
 
     /**
-     * If set to <code>false</code>, a new stream won't be created if no stream matches the given key.
-     * <br>
-     * Equivalent to <code>NOMKSTREAM</code> in the Redis API.
+     * Optional arguments to {@link StreamBaseCommands#xadd(String, Map, StreamAddOptions)}
+     *
+     * @see <a href="https://redis.io/commands/xadd/">redis.io</a>
      */
-    private final Boolean makeStream;
+    @Builder
+    public static final class StreamAddOptions {
+        public static final String NO_MAKE_STREAM_REDIS_API = "NOMKSTREAM";
+        public static final String ID_WILDCARD_REDIS_API = "*";
+
+        /** If set, the new entry will be added with this <code>id</code>. */
+        private final String id;
+
+        /**
+         * If set to <code>false</code>, a new stream won't be created if no stream matches the given
+         * key. <br>
+         * Equivalent to <code>NOMKSTREAM</code> in the Redis API.
+         */
+        private final Boolean makeStream;
+
+        /** If set, the add operation will also trim the older entries in the stream. */
+        private final StreamTrimOptions trim;
+
+        /**
+         * Converts options for Xadd into a String[].
+         *
+         * @return String[]
+         */
+        public String[] toArgs() {
+            List<String> optionArgs = new ArrayList<>();
+
+            if (makeStream != null && !makeStream) {
+                optionArgs.add(NO_MAKE_STREAM_REDIS_API);
+            }
+
+            if (trim != null) {
+                optionArgs.addAll(trim.getRedisApi());
+            }
+
+            if (id != null) {
+                optionArgs.add(id);
+            } else {
+                optionArgs.add(ID_WILDCARD_REDIS_API);
+            }
 
-    /** If set, the add operation will also trim the older entries in the stream. */
-    private final StreamTrimOptions trim;
+            return optionArgs.toArray(new String[0]);
+        }
+    }
 
+    /**
+     * Optional arguments for {@link StreamBaseCommands#xtrim(String, StreamTrimOptions)}
+     *
+     * @see <a href="https://redis.io/commands/xtrim/">redis.io</a>
+     */
     public abstract static class StreamTrimOptions {
+
+        public static final String TRIM_MAXLEN_REDIS_API = "MAXLEN";
+        public static final String TRIM_MINID_REDIS_API = "MINID";
+        public static final String TRIM_EXACT_REDIS_API = "=";
+        public static final String TRIM_NOT_EXACT_REDIS_API = "~";
+        public static final String TRIM_LIMIT_REDIS_API = "LIMIT";
+
         /**
          * If <code>true</code>, the stream will be trimmed exactly. Equivalent to <code>=</code> in the
          * Redis API. Otherwise, the stream will be trimmed in a near-exact manner, which is more
          * efficient, equivalent to <code>~</code> in the Redis API.
          */
-        protected boolean exact;
+        protected Boolean exact;
```

Comment:
Maybe we should use 3 state enum here instead

Reply:
yes missing (not specified), true, false

Reply:
I think what we have is fine...